projects/media-viewer/src/lib/store/reducers/bookmarks.reducer.ts

```diff
@@ -54,6 +54,38 @@ export function bookmarksReducer (state = initialBookmarksState,
         loaded: true
       }
     }
+
+    case fromBookmarks.DELETE_BOOKMARK_SUCCESS: {
+      const id: String = action.payload;
+      const bookmarks = state.bookmarks.filter(bookmark => bookmark.id !== id);
+      return {
+        ...state,
+        bookmarks,
+        loading: false,
+        loaded: true
+      }
+    }
+
+    case fromBookmarks.UPDATE_BOOKMARK_SUCCESS: {
+      const bookmark: Bookmark = action.payload;
+      const name = bookmark.name;
+      const bookmarks = state.bookmarks.map(bmark => {
```

Comment:
why not just filter like you're doing with delete, and insert the update one in
const bookmarks = [ ...state.bookmarks.filter(bookmark => bookmark.id !== id), bookmark]

Reply:
Because I need to update the name of one of the bookmarks

Reply:
yes, the bookmark you insert has the updated name

Reply:
Oh of course, ok I will change it

Reply:
Oh I remember why I did it like this... const bookmarks = [ ...state.bookmarks.filter(bookmark => bookmark.id !== id), bookmark] changes the position of the edited bookmark to the end of the array